src/Components/WebAssembly/WebAssembly/src/HotReload/WebAssemblyHotReload.cs

```diff
@@ -34,13 +34,9 @@ internal static async Task InitializeAsync()
         if (Environment.GetEnvironmentVariable("__ASPNETCORE_BROWSER_TOOLS") == "true")
         {
             // Attempt to read previously applied hot reload deltas if the ASP.NET Core browser tools are available (indicated by the presence of the Environment variable).
-            // The agent is injected in to the hosted app and can serve this script that can provide results from local-storage .
+            // The agent is injected in to the hosted app and can serve this script that can provide results from local-storage.
             // See https://github.com/dotnet/aspnetcore/issues/37357#issuecomment-941237000
-
-            var jsObjectReference = (IJSUnmarshalledObjectReference)(await DefaultWebAssemblyJSRuntime.Instance.InvokeAsync<IJSObjectReference>("import", "/_framework/blazor-hotreload.js"));
-#pragma warning disable CS0618 // Type or member is obsolete
-            await jsObjectReference.InvokeUnmarshalled<Task<int>>("receiveHotReload");
-#pragma warning restore CS0618 // Type or member is obsolete
+            await InternalJSImportMethods.Instance.InitHotReloadAsync("/_framework/blazor-hotreload.js");
```

Comment:
there is `JSHost.ImportAsync()` which could replace this.

Reply:
That's interesting. If we did that, just a quick reminder to verify that the semantics are equivalent (e.g., in how URLs are resolved against `<base ref>`, though maybe this is a root-relative URL and hence does not vary anyway).

Reply:
`ImportAsync` is relative to `dotnet.js` so I guess it would be `./blazor-hotreload.js`

Reply:
I've verified that `JSHost.ImportAsync()` has the same URL resolution behavior in this scenario.